boards/arm/stm32f411e_disco/stm32f411e_disco_defconfig

```diff
@@ -0,0 +1,51 @@
+CONFIG_ARM=y
+CONFIG_BOARD_STM32F411E_DISCO=y
+CONFIG_SOC_SERIES_STM32F4X=y
+CONFIG_SOC_STM32F411XE=y
+# 100MHz system clock (highest value to get a precise USB clock should be 96MHz)
+CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=100000000
+CONFIG_SYS_CLOCK_TICKS_PER_SEC=1000
+
+# Enable MPU
+CONFIG_STM32_ARM_MPU_ENABLE=y
+
+# Serial Drivers
+CONFIG_SERIAL=y
+CONFIG_SERIAL_HAS_DRIVER=y
+CONFIG_UART_INTERRUPT_DRIVEN=y
+CONFIG_UART_STM32=y
+# enable USART2 - passthrough to STLINK v2 connector
+CONFIG_UART_STM32_PORT_2=y
+# enable console on this port by default
+CONFIG_CONSOLE=y
+CONFIG_UART_CONSOLE=y
+
+# enable pinmux
+CONFIG_PINMUX=y
+CONFIG_PINMUX_STM32=y
+
+# enable GPIOs
+CONFIG_GPIO=y
+CONFIG_GPIO_STM32=y
+CONFIG_GPIO_STM32_PORTA=y
+CONFIG_GPIO_STM32_PORTB=y
+CONFIG_GPIO_STM32_PORTC=y
+CONFIG_GPIO_STM32_PORTD=y
+CONFIG_GPIO_STM32_PORTE=y
```

Comment:
Currently, on stm32f4xx, GPIO activation is done under arch/arm/soc/st_stm32/stm32f4/Kconfig.defconfig.series.

Reply:
Sorry, maybe I misunderstood how GPIO config works... but in soc you declare all options for it (chip) and in board you enable the wired GPIO's, is that ok?
So, I used an already created soc, and put on a board with those GPIOs wired to headers.
But if the right way is to remove this from board and modify soc, let me know and I will delete it (soc modification is better to be done in a separate pull request, I think)
Thank you!

Reply:
Actually, activation could also be done in Kconfig.defconfig.xxx files, and this is the way it is done today on F4 for GPIO in soc. This allows to factorize some code that would need to be copied in each board _defconfig file.
Maybe this is not the best solution as we can argue that GPIO activation is strictly board dependant, but let's keep it coherent across the family.

Reply:
OK, I will modify right now, but will introduce same issue that tries to solve #4300 (for other board and soc)
On the other way, if all GPIO ports are enabled for all F4 family, it's not a problem with a board that uses a soc with not all GPIO present on the chip? (48 and 64 pin configurations, for example)

Reply:
This is right. As mentioned already this might not be the best solution, but I try to keep things coherent for now. And indeed, activating a GPIO not present on a SoC (as all SoCs do not support same number of GPIO ports) will issue an error.

A clean an optimized way to do it should be to do all supported GPIO activation by default in soc with the finest granularity (Kconfig.def_config.stm32f4xxyy files depending on SoC variant and SoC package). Then, depending on application, GPIO could be deactivated in board/xx_defconfig files. 
Besides, this would be less error prone than activating minimum GPIOs in boards and debug hours a new use case to discover in the end that GPIO port was not activated (as this is the case today).

Though, this should be configured by device tree in short term, so not I'm not quite willing to rework this for now.

Reply:
I agree, with a rework in the Kconfig.def_config.stm32f4xxyy files we can keep it coherent, going to the finest granularity, but actually I see no way of doing so without breaking de boards with lower pin-count parts, because this board's leds are on PORT D, so, it's needed for the basic samples to work.

Maybe we can go that way for now? Or wait untill we can configure GPIO by device tree (any info on how to do this?)
Another option do you recommend?

Thank you for your time!